napari/_qt/dialogs/qt_package_installer.py

```diff
@@ -0,0 +1,419 @@
+                if i == 0:  # first in queue, currently running
+                    end_process()
+                else:  # still pending, just remove from queue
+                    self._queue.remove(item)
+                return
+        msg = f"No job with id {job_id}. Current queue:\n - "
+        msg += "\n - ".join(
+            [
+                f"{item.ident} -> {item.executable()} {item.arguments()}"
+                for item in self._queue
+            ]
+        )
+        raise ValueError(msg)
+
+    def waitForFinished(self, msecs: int = 10000) -> bool:
+        """Block and wait for all jobs to finish.
+
+        Parameters
+        ----------
+        msecs : int, optional
+            Time to wait, by default 10000
+        """
+        while self.hasJobs():
+            super().waitForFinished(msecs)
+        return True
+
+    def hasJobs(self) -> bool:
+        """True if there are jobs remaining in the queue."""
+        return bool(self._queue)
+
+    def set_output_widget(self, output_widget: QTextEdit):
+        if output_widget:
+            self._output_widget = output_widget
+
+    # -------------------------- Private methods ------------------------------
+    def _log(self, msg: str):
+        log.debug(msg)
+        if self._output_widget:
+            self._output_widget.append(msg)
+
+    def _build_queue_item(
+        self,
+        task_handler: TaskHandler,
+        action: ActionType,
+        pkgs: Sequence[str],
+        prefix: Optional[str] = None,
+        origins: Sequence[str] = (),
+        **kwargs,
+    ) -> AbstractInstallerTask:
+        if task_handler == "pip":
+            InstallerAction = PipInstallerTask
+        elif task_handler == "conda":
+            InstallerAction = CondaInstallerTask
+        else:
+            raise ValueError(f"Handler {task_handler} not recognized!")
+        return InstallerAction(
+            pkgs=pkgs, action=action, origins=origins, prefix=prefix, **kwargs
+        )
+
+    def _queue_item(self, item: AbstractInstallerTask) -> JobId:
+        self._queue.append(item)
+        self._process_queue()
+        return item.ident
+
+    def _process_queue(self):
+        if not self._queue:
+            self.allFinished.emit()
+            return
+        task = self._queue[0]
+        self.setProgram(str(task.executable()))
+        self.setProcessEnvironment(task.environment())
+        self.setArguments([str(arg) for arg in task.arguments()])
+        # this might throw a warning because the same process
+        # was already running but it's ok
+        self._log(f"Starting '{self.program()}' with args {self.arguments()}")
+        self.start()
+
+    def _on_process_finished(
+        self, exit_code: int, exit_status: QProcess.ExitStatus
+    ):
+        try:
+            current = self._queue[0]
+        except IndexError:
+            current = None
+        if (
+            current
+            and current.action == "uninstall"
```

Comment:
I would rather we had also some "enums" for these actions :) 

See example above